ci_framework/roles/dlrn_promote/tasks/check_for_previous_promotions.yml

```diff
@@ -0,0 +1,42 @@
+---
+- name: Check reports from DLRN
+  ansible.builtin.shell:
+    cmd: >-
+      set -o pipefail;
+      export SSL_CA_BUNDLE={{ cifmw_dlrn_promote_ssl_ca_bundle }};
+      dlrnapi --url {{ cifmw_repo_setup_dlrn_api_url }}
+      {% if cifmw_dlrn_promote_dlrnapi_host_principal is defined and cifmw_dlrn_promote_kerberos_auth|bool -%}
+      --server-principal {{ cifmw_dlrn_promote_dlrnapi_host_principal }} --auth-method kerberosAuth --force-auth
+      {% endif -%}
+      promotion-get
+      {% if (cifmw_repo_setup_full_hash is defined) and (cifmw_repo_setup_full_hash | length > 0) -%}
+      --agg-hash {{ cifmw_repo_setup_full_hash }}
+      {% endif -%}
+      {% if (cifmw_repo_setup_extended_hash is defined) and (cifmw_repo_setup_extended_hash | length > 0) -%}
+      --extended_hash {{ cifmw_repo_setup_extended_hash }}
+      {% endif -%}
+      {% if (cifmw_repo_setup_commit_hash is defined) and (cifmw_repo_setup_commit_hash | length > 0) -%}
+      --commit_hash {{ cifmw_repo_setup_commit_hash }}
+      {% endif -%}
+      {% if (cifmw_repo_setup_distro_hash is defined) and (cifmw_repo_setup_distro_hash | length > 0) -%}
+      --distro_hash {{ cifmw_repo_setup_distro_hash }}
+      {% endif -%}
+      --promote-name {{ cifmw_dlrn_promote_promotion_target }}
+      | tee -a {{ cifmw_dlrn_promote_workspace }}/cifmw_dlrn_promote_hash_promote_target_output.txt
+  environment: |
+    {{ zuul | zuul_legacy_vars | combine({
+      'DLRNAPI_PASSWORD': dlrnapi.password,
+      'DLRNAPI_USERNAME': dlrnapi_user
+      }) }}
+  changed_when: true
+
+- name: Get the the DLRN output results
+  ansible.builtin.slurp:
+    path: "{{ cifmw_dlrn_promote_workspace }}/cifmw_dlrn_promote_hash_promote_target_output.txt"
```

Comment:
Is the output a bare txt or is it json/yaml? Cause I see comparisons with `'[]'` and in that case the way to go is to compare the data itself with a `| length > 0` after properly calling `from_json` or `from_yaml`

Reply:
It should return a list. fixed it.